deps/kms/src/plugins/sample.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use aes_gcm::{aead::Aead, Aes256Gcm, Key, KeyInit, Nonce};
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use crypto::WrapType;
+use rand::RngCore;
+use zeroize::Zeroizing;
+
+use crate::{Datakey, KMS};
+
+const IV: &[u8] = b"unique nonce";
+
+/// Only for sample
+pub const EXAMPLE_CMK: &[u8] = &[
+    217, 155, 119, 5, 176, 186, 122, 22, 130, 149, 179, 163, 54, 114, 112, 176, 221, 155, 55, 27,
+    245, 20, 202, 139, 155, 167, 240, 163, 55, 17, 218, 234,
+];
+
+/// A fake a KMS implementation
+pub struct SampleKms;
+
+#[async_trait]
+impl KMS for SampleKms {
+    async fn generate_data_key(&mut self) -> Result<Box<dyn Datakey>> {
+        Ok(Box::<SampleDatakey>::default())
+    }
+}
+
+/// Fake Datakey implementation
+pub struct SampleDatakey {
+    /// The `Zeroizing` here will help to flush the memory when the
+    /// [`Datakey`] object is dropped
+    key: Zeroizing<Vec<u8>>,
+}
+
+impl Default for SampleDatakey {
+    /// Here we simulate behavior of KMS. A random datakey will be generated
+    fn default() -> Self {
+        let mut key = Zeroizing::new(Vec::new());
+        key.resize(32, b' ');
+        rand::thread_rng().fill_bytes(&mut key);
+        Self { key }
+    }
+}
+
+#[async_trait]
+impl Datakey for SampleDatakey {
+    async fn encrypt(&mut self, plaintext: &[u8]) -> Result<Vec<u8>> {
+        let encryption_key = Key::<Aes256Gcm>::from_slice(&self.key);
+        let cipher = Aes256Gcm::new(encryption_key);
+        let nonce = Nonce::from_slice(IV);
```

Comment:
This is kind of a bad example, because the IV needs to be returned together with the ciphertext.
And decrypt needs to receive the IV.